### PR TITLE
WIP: StencilJS renderer

### PR DESCRIPTION
**Pull request by natemoo-re**
# Draft PR to hash out some `custom-element` issues

## Changes

This is an experimental PR to add SSR support for [Stencil](https://stenciljs.com) custom elements.

## Testing

- [ ] Tests are passing
- [ ] Tests updated where necessary

## Docs

- [ ] Docs / READMEs updated
- [ ] Code comments added where helpful

### CODE REVIEW

File: packages/astro/src/compiler/codegen/index.ts [hunk 1]
```diff
@@ -136,10 +136,19 @@ interface GetComponentWrapperOptions {
 const PlainExtensions = new Set(['.js', '.jsx', '.ts', '.tsx']);
 /** Generate Astro-friendly component import */
 function getComponentWrapper(_name: string, { url, importSpecifier }: ComponentInfo, opts: GetComponentWrapperOptions) {
+  const [name, kind] = _name.split(':');
+
+  // Special flow for custom elements
+  if (isCustomElementTag(name)) {
```
[the remark this answers is not in the document]
Comment (matthewp) [reply packages/astro/src/compiler/codegen/index.ts:142]: In the Lit issue we did figure out that WC devs want to use the tag name and not PascalCase so I think this is good to include.

**Comment by FredKSchott** [comment]
Update: @eyelidlessness is working on this and posting status updates in Discord. We can probably close this PR once that one is opened https://discord.com/channels/830184174198718474/852973466758938696/853719552818675778

**Comment by matthewp** [comment]
@FredKSchott They are working on Solid support I think, right? This is for Stencil.

**Comment by FredKSchott** [comment]
Doh, got my frameworks mixed up

**Comment by matthewp** [comment]
#587 is in. Would be nice to see if that takes care of the needs for Stencil. Much of what that PR did was inspired by some of the changes here.

**Comment by natemoo-re** [comment]
Closing as #587 landed most of these changes. Will revisit this when possible.

**Pull request closed without merging**